### DISCUSSION TITLE
Error starting CKAN database

### DISCUSSION THREAD

**Discussion by rogermsarmento** [ckan/ckan#6825] [Q&A]
**CKAN version**: 2.9
**Ubuntu version**: 20.04
**Python version**: 3.8.10

**CKAN Version if known (or site URL)**
CKAN 2.5

Please describe the expected behaviour
When running the commmand sudo ckan db init, the database should be successfully initialized.

Please describe the actual behaviour
When running the command sudo ckan db init the following error occurs:

```
dgti@dgti-VirtualBox:/etc/ckan/default$ sudo ckan db init
Traceback (most recent call last):
  File "/usr/lib/ckan/default/bin/ckan", line 11, in <module>
    load_entry_point('ckan', 'console_scripts', 'ckan')()
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 781, in main
    with self.make_context(prog_name, args, **extra) as ctx:
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 700, in make_context
    self.parse_args(ctx, args)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 115, in parse_args
    result = super(ExtendableGroup, self).parse_args(ctx, args)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1212, in parse_args
    rest = Command.parse_args(self, ctx, args)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1048, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1630, in handle_parse_result
    value = invoke_param_callback(self.callback, ctx, self, value)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 123, in invoke_param_callback
    return callback(ctx, param, value)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 125, in _init_ckan_config
    _add_ctx_object(ctx, value)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 134, in _add_ctx_object
    ctx.obj = CtxObject(path)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 55, in __init__
    self.config = ckan_cli.load_config(conf)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/__init__.py", line 120, in load_config
    config_loader = CKANConfigLoader(filename)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/__init__.py", line 24, in __init__
    self._create_config_object()
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/__init__.py", line 52, in _create_config_object
    self._update_config()
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/__init__.py", line 36, in _update_config
    options = self.parser.options(self.section)
  File "/usr/lib/python3.8/configparser.py", line 675, in options
    raise NoSectionError(section) from None
configparser.NoSectionError: No section: 'app:main'
```

What steps can be taken to reproduce the issue?
I am following the [CKAN installation instructions](https://docs.ckan.org/en/2.9/maintaining/installing/install-from-package.html).

1. Install the CKAN package - ok
2. Install and configure PostgreSQL - ok
3. Install and configure Solr - ok
4. Update the configuration and initialize the database - **Erro** in "sudo ckan db init"
5. Start the Web Server and restart Nginx - **Erro** 

![image](https://user-images.githubusercontent.com/40036396/165525121-9ff17bd0-2951-42b4-82cd-2ee863d4a412.png)

6. You’re done! - **fail**
7. 
![image](https://user-images.githubusercontent.com/40036396/165525511-2dbd14fd-6244-450d-ada8-078522bdcc9b.png)

**Describe the bug**
I installed 2.9 from package and every time i run the "sudo ckan db init" i get the following error below:

Whats happening?

**Comment by TomeCirun** [comment]
@rogermsarmento do you have `[app:main]`  in your `.ini` file ?

![image](https://user-images.githubusercontent.com/72216462/165525753-f4ae23e7-2f89-490a-9599-c33630201392.png)

**Reply by rogermsarmento** [reply]
Hi, @TomeCirun 

/etc/ckan/default/ckan.ini was never created. 
Inside the default folder there were these files:

![image](https://user-images.githubusercontent.com/40036396/165527700-26ae6d35-4a04-4bc7-b004-adb950c3471a.png)



ckan.ini is a copy of ckan-uwsgi.ini with some changes.

![image](https://user-images.githubusercontent.com/40036396/165528628-48b75cee-41af-44d1-9f6a-2cb64d16780f.png)



Any idea?

**Comment by TomeCirun** [comment]
You can create your config file: `ckan generate config /etc/ckan/default/ckan.ini` 

visit https://docs.ckan.org/en/2.9/maintaining/configuration.html for more info.

**Reply by rogermsarmento** [reply]
Hi @TomeCirun, thanks a lot for the help. I'm still very noob in ckan. 

But I thought the ckan.ini file was generated automatically after following the site's tutotial

[https://docs.ckan.org/en/2.9/maintaining/installing/install-from-package.html](url)

**Reply by rogermsarmento** [reply]
@TomeCirun Could you show me what the ckan.ini file should look like?

**Reply by tino097** [reply]
https://github.com/ckan/ckan/blob/master/ckan/config/deployment.ini_tmpl

using the command suggested by @TomeCirun you should have the `development.ini` generated

**Reply by rogermsarmento** [reply]
Hi @tino097, thanks for your help!

I followed all the steps in the link [[https://docs.ckan.org/en/2.9/maintaining/installing/install-from-package.html](https://github.com/ckan/ckan/discussions/url)](url). 

but I got this result:

![image](https://user-images.githubusercontent.com/40036396/166453096-cbfec492-34f4-4212-8a03-919ab04b1d6d.png)

when I run the command `ckan generate config /etc/ckan/default/ckan.ini` i get this:

![image](https://user-images.githubusercontent.com/40036396/166453521-7bbb4caf-ad04-4d1b-a260-5c6c11b3aff7.png)

I really need help.

**Reply by rogermsarmento** [reply]
@tino097 e @TomeCirun I created the ckan.ini file through the ubuntu terminal.

![image](https://user-images.githubusercontent.com/40036396/166461017-171d9c71-1197-4b84-b3ab-dd18a46f7a5b.png)

I will configure the file:

![image](https://user-images.githubusercontent.com/40036396/166461087-733a02f1-e0e8-4d9e-8cc2-d644255b7c84.png)

I restarted the server with the `sudo service tomcat9 restart` and initialized the CKAN database with the `sudo ckan db init`. 

**return:**

> dgti@dgti-VirtualBox:/etc/ckan/default$ sudo ckan db init
> Traceback (most recent call last):
>   File "/usr/lib/python3.8/configparser.py", line 789, in get
>     value = d[option]
>   File "/usr/lib/python3.8/collections/__init__.py", line 898, in __getitem__
>     return self.__missing__(key)            # support subclasses that define __missing__
>   File "/usr/lib/python3.8/collections/__init__.py", line 890, in __missing__
>     raise KeyError(key)
> KeyError: 'use'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/usr/lib/ckan/default/bin/ckan", line 11, in <module>
>     load_entry_point('ckan', 'console_scripts', 'ckan')()
>   File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 829, in __call__
>     return self.main(*args, **kwargs)
>   File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 781, in main
>     with self.make_context(prog_name, args, **extra) as ctx:
>   File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 700, in make_context
>     self.parse_args(ctx, args)
>   File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 115, in parse_args
>     result = super(ExtendableGroup, self).parse_args(ctx, args)
>   File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1212, in parse_args
>     rest = Command.parse_args(self, ctx, args)
>   File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1048, in parse_args
>     value, args = param.handle_parse_result(ctx, opts, args)
>   File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1630, in handle_parse_result
>     value = invoke_param_callback(self.callback, ctx, self, value)
>   File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 123, in invoke_param_callback
>     return callback(ctx, param, value)
>   File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 125, in _init_ckan_config
>     _add_ctx_object(ctx, value)
>   File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 134, in _add_ctx_object
>     ctx.obj = CtxObject(path)
>   File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 55, in __init__
>     self.config = ckan_cli.load_config(conf)
>   File "/usr/lib/ckan/default/src/ckan/ckan/cli/__init__.py", line 120, in load_config
>     config_loader = CKANConfigLoader(filename)
>   File "/usr/lib/ckan/default/src/ckan/ckan/cli/__init__.py", line 24, in __init__
>     self._create_config_object()
>   File "/usr/lib/ckan/default/src/ckan/ckan/cli/__init__.py", line 57, in _create_config_object
>     schema, path = self.parser.get(self.section, u'use').split(u':')
>   File "/usr/lib/python3.8/configparser.py", line 792, in get
>     raise NoOptionError(option, section)
> configparser.NoOptionError: No option 'use' in section: 'app:main'

Any idea?

**Reply by TomeCirun** [reply]
try to set `use = egg:ckan`  right after `[app:main]`

**Reply by rogermsarmento** [reply]
Hi @TomeCirun 

Done:

![image](https://user-images.githubusercontent.com/40036396/166679759-78c632a0-de27-4843-98dc-8d8d45ed9ba7.png)

result of `sudo ckan db init`:

```
Traceback (most recent call last):
  File "/usr/lib/ckan/default/bin/ckan", line 11, in <module>
    load_entry_point('ckan', 'console_scripts', 'ckan')()
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 781, in main
    with self.make_context(prog_name, args, **extra) as ctx:
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 700, in make_context
    self.parse_args(ctx, args)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 115, in parse_args
    result = super(ExtendableGroup, self).parse_args(ctx, args)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1212, in parse_args
    rest = Command.parse_args(self, ctx, args)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1048, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 1630, in handle_parse_result
    value = invoke_param_callback(self.callback, ctx, self, value)
  File "/usr/lib/ckan/default/lib/python3.8/site-packages/click/core.py", line 123, in invoke_param_callback
    return callback(ctx, param, value)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 125, in _init_ckan_config
    _add_ctx_object(ctx, value)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 134, in _add_ctx_object
    ctx.obj = CtxObject(path)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/cli.py", line 55, in __init__
    self.config = ckan_cli.load_config(conf)
  File "/usr/lib/ckan/default/src/ckan/ckan/cli/__init__.py", line 121, in load_config
    loggingFileConfig(filename)
  File "/usr/lib/python3.8/logging/config.py", line 71, in fileConfig
    formatters = _create_formatters(cp)
  File "/usr/lib/python3.8/logging/config.py", line 104, in _create_formatters
    flist = cp["formatters"]["keys"]
  File "/usr/lib/python3.8/configparser.py", line 960, in __getitem__
    raise KeyError(key)
KeyError: 'formatters'

```
This problem should be very simple to solve, but it's killing me.

**Reply by rogermsarmento** [reply]
@TomeCirun @tino097 Thanks a lot for the help. 

I can't solve the problem so I created another VM and repeated all steps. This time the ckan.ini file was complete and worked.
